packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -454,6 +460,7 @@ class DatePicker extends React.Component {
             onChange={this.handleChange}
             onChangeRaw={this.handleChangeRaw}
             onClickOutside={this.handleOnClickOutside}
+            onDismiss={onDismiss}
```

Comment:
onClickOutside should be removed and passed like:
```suggestion
            onDismiss={onDismiss || onClickOutside}
```

Reply:
I think to keep it as a passive change we must allow both of them to be passed simultaneously. As if we give precedence to onDismiss, it is breaking the mobile view as that being a wrapped calendar requires `onClickOutside` event handler.

Reply:
@pranav300 Wouldn't you pass onDismiss to the onClickOutside for now to remain passive?

Reply:
`onDismiss` is being used in [setOpen](https://github.com/cerner/terra-framework/pull/1224/files#diff-0eaf044a5babfc8ff56c4b57c5220edbR471).

But yeah removing [this](https://github.com/cerner/terra-framework/pull/1224/files#diff-0eaf044a5babfc8ff56c4b57c5220edbL504), allows us to have that precedence.

Reply:
I thought was we will leave `onClickOutside` the way it is today where it is called only when clicking outside the picker to close it. We are adding `onDismiss`, which will get called whenever the picker closed (whether it be clicking outside or selecting a day). I was thinking to separate `onDismiss` with `onClickOutside` and not have them interfere with one another. Thoughts?

Reply:
@benbcai I do agree that providing precedence to `onDismiss` will make it a non-passive change. Therefore, I have reverted that commit here https://github.com/cerner/terra-framework/pull/1224/commits/9d8be2a715656afc67046abb501756e8ef610399. 

I think we will just have to wait to handle this in #1222.